examples/snake-cli/coordinate.rs

```diff
@@ -0,0 +1,60 @@
+use crate::direction::Direction;
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Coordinate {
+    pub x: usize,
+    pub y: usize,
+}
+impl Coordinate {
+    /// This function returns true if it causes an over-/underflow upon which the snake should die
+    pub fn transform(&mut self, direction: Direction, step: usize) -> bool {
+        match direction {
+            // Since this could cause an over-/underflow it's safer to use checked_sub
+            Direction::Up => {
+                if let Some(x) = usize::checked_sub(self.y, step) {
+                    self.y = x;
+                    false
+                } else {
+                    true
+                }
+            }
+            Direction::Right => {
+                if let Some(x) = usize::checked_add(self.x, step) {
+                    self.x = x;
+                    false
+                } else {
+                    true
+                }
+            }
+            Direction::Down => {
+                if let Some(x) = usize::checked_add(self.y, step) {
+                    self.y = x;
+                    false
+                } else {
+                    true
+                }
+            }
+            Direction::Left => {
+                if let Some(x) = usize::checked_sub(self.x, step) {
+                    self.x = x;
+                    false
+                } else {
+                    true
+                }
+            }
+        }
+    }
+    pub fn new_transform(
+        coordinate: Coordinate,
+        direction: Direction,
+        step: usize,
+    ) -> (Self, bool) {
+        let mut clone = coordinate;
```

Comment:
You could have the `mut` at line 47

Reply:
I think you're talking about the formatting. Yeah here it is kind of unnecessary how many new lines there are but overall i think cargo fmt does a good job for autoformatting